poly-commit/src/linear_codes/mod.rs

```diff
@@ -553,7 +553,7 @@ where
     H::Output: Into<C::Leaf>,
     C::Leaf: Default + Clone + Send,
 {
-    let ext_mat_cols = ext_mat.cols();
+    let ext_mat_cols = ext_mat.cols().clone();
```

Comment:
> So we should avoid cloning and rearranging as much as we can. This PR is trying to solve that!

Huh?

Reply:
This one is inevitable, unfortunately. `H::evaluate` consumes the vector as far as I understood. Notice that we were cloning it previously too, but it was not explicit here.

Reply:
But with returning a reference, now we just clone `t` vectors here
https://github.com/HungryCatsStudio/poly-commit/blob/ddfd74d7669e405d15fe5285d1fc0f6e0b5cbd92/poly-commit/src/linear_codes/mod.rs#L599-L602

Reply:
In fact, if we mange to derive `Copy` for `Vec<F>` we can avoid cloning here too, I guess

Reply:
@autquis I figured out that you can actually just call `.borrow()` and it works like a charm. See e20f67e8122b26f1adea8847d491c65d41588ab1

Reply:
@mmagician Nice!
Also, let me know if I can still help with this PR!